R/train.R

```diff
@@ -31,26 +31,36 @@
 train = function(learner, task, subset, weights = NULL) {
   learner = checkLearner(learner)
   assertClass(task, classes = "Task")
-  if (missing(subset) || is.null(subset)) {
+  if (missing(subset))
+    subset = NULL
```

Comment:
I assume we break stuff if we let subset default to `NULL`?

Reply:
I can set the default to `NULL` and see what happens, I was just trying to keep the UI changes as small as possible (since the `missing(subset) || is.null(subset)` bit seemed intentional).